src/main/java/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,10 @@
+package operation;
+
+import service.BalanceService;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void changeBalance(BalanceService balanceService, String item, Integer quantity) {
```

Comment:
we should not pass another service in method params, please create a field and use a constructor